Sources/MongoSwift/MongoClient.swift

```diff
@@ -343,6 +343,7 @@ public class MongoClient {
      *   - filter: Optional `Document` specifying a filter that the listed databases must pass. This filter can be based
      *     on the "name", "sizeOnDisk", "empty", or "shards" fields of the output.
      *   - session: Optional `ClientSession` to use when executing this command.
+     *   - authorizedDatabases: Optional `Bool` disabling will have mongodb return an error if user in unauthorized to list databases.
```

Comment:
I think we should definitely put it in an options struct, both for forward compatibility and consistency with the rest of the driver. `filter` can stay as-is for the reasons you mention.

Reply:
Should `nameOnly` be in the options object too? and if so then I feel like we drop `listDatabaseNames()` since we'll be taking in an options object from the user:
```swift
/// Options for "listDatabases" operations
public struct ListDatabasesOperationOptions {
    /// Specifies whether to only return databases for which the user has privileges.
    public let authorizedDatabases: Bool?
    /// Specifies whether to only return the list of database names
    public let nameOnly: Bool?
}
```

Reply:
That is a good question, which we spent a while discussing back when I think @patrickfreed initially implemented the database enumeration spec.

I think we should continue to keep `nameOnly` hidden from the user and just expose it via the separate API methods.

IMO the biggest advantage of the current approach is that, since we use separate methods to accomplish nameOnly vs not-nameOnly, we can use different return types for them -- `listDatabaseNames` returns an `[String]` while `listDatabases` returns a `DatabaseSpecification`.

Were we to combine these into one method, we'd have to return the same type in both situations. This type would have to be more flexible to support both cases, and would cause us to lose specificity / information about what information should be present when. It would probably look like `DatabaseSpecification` but have all fields except `name` be optionals. (Alternatively we could make the `ListDatabasesResults` enum public.)
Consider a user who doesn't pass in `nameOnly` to the method. We can guarantee that we will have the `sizeOnDisk` field available in the object we return when they set that to true, yet it would somewhat confusingly to them be an optional.
Alternatively, if we returned a `ListDatabasesResults`, they would have to switch on the returned value and handle the case where `.names` is returned, even though we know that's actually impossible.
Splitting up the methods allows us to be more precise in both cases about the present information.

Reply:
Another rationale in addition to all that is that `nameOnly: true` has performance implications server side due to having different locking requirements, so it makes sense to communicate that at the type level.